src/python/turicreate/data_structures/sarray.py

```diff
@@ -2902,6 +2902,12 @@ def append(self, other):
         Rows: 6
         [1, 2, 3, 4, 5, 6]
         """
+        if len(other) == 0:
```

Comment:
Let's move these under the the  `if self.dtype != other.dtype:` statement.  The reason for this is that len(...) may trigger the materialization of an unmaterialized SArray, potentially expensive operation.  Thus let's only do it if we have to.

Reply:
@hoytak  @TobyRoseman  Please resolve the conflicting files issues in master
 File name : src/python/turicreate/test/test_sarray.py .